### PR TITLE
Docs: Go SDK

### PR DESCRIPTION
**Pull request by DanielHougaard**
# Description 📣

Docs for our new Go SDK. 

## Type ✨

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Documentation

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Review by maidul98 — DISMISSED** [review]
looks good to me! Just small nit about changing from site url to host

File: docs/sdks/languages/go.mdx [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+
+If you're working with Go Lang, the official [Infisical Go SDK](https://github.com/infisical/go-sdk) package is the easiest way to fetch and work with secrets for your application.
+
+- [Package](https://pkg.go.dev/github.com/infisical/go-sdk)
+- [Github Repository](https://github.com/infiscial/go-sdk)
+
+## Basic Usage
+
+```go
+import (
```
Comment (ShubhamPalriwala) [inline docs/sdks/languages/go.mdx:17]: Let's add the required `package main` as it currently throws an error without it as we try to run

File: docs/sdks/languages/go.mdx [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+  SecretKey:     "NEW_SECRET_KEY",
+  SecretValue:   "NEW_SECRET_VALUE",
+  SecretComment: "This is a new secret",
+})
+```
+
+Create a new secret in Infisical.
+
+#### Parameters
```
Comment (ShubhamPalriwala) [inline docs/sdks/languages/go.mdx:334]: Can we add the `NewSkipMultilineEncoding` for secret creation as well? We plan to do that in the UI itelf + it's also supported in the backend i think

**Review by ShubhamPalriwala — COMMENTED** [review]
Great work! The docs are clean & work smoothly! Tested all methods except the auths (tried only universal) 🚀 and work as expected ✅ 

Just added a couple of small notes, please see if they make sense to you as well.

File: docs/sdks/languages/go.mdx (hunk 2 above)
Comment (maidul98) [reply docs/sdks/languages/go.mdx:334]: Good idea, we can add that in on a later iteration. Not a hard blocker for this PR

**Review by maidul98 — APPROVED** [review]

**Pull request merged**